core/metacling/src/TCling.cxx

```diff
@@ -1236,13 +1236,30 @@ static void RegisterCxxModules(cling::Interpreter &clingInterp)
       clang::CompilerInstance &CI = *clingInterp.getCI();
       GlobalModuleIndex *GlobalIndex = nullptr;
       // Conservatively enable platform by platform.
-      bool supportedPlatform = false;
+      bool supportedPlatform =
+#ifdef R__LINUX
+         true
+#elif defined(R__MACOSX)
+         true
+#else // Windows
+         false
+#endif
+         ;
       // Allow forcefully enabling the GMI.
-      const char *experimentalGMI = gSystem->Getenv("ROOT_EXPERIMENTAL_GMI");
-      if (experimentalGMI && strcmp(experimentalGMI,"false") != 0)
+      llvm::Optional<std::string> EnvEnGMI = llvm::sys::Process::GetEnv("ROOT_EXPERIMENTAL_GMI");
+      if (EnvEnGMI.hasValue() && ROOT::FoundationUtils::ConvertEnvValueToBool(*EnvEnGMI))
          supportedPlatform = true;
 
-      if (supportedPlatform && !gSystem->Getenv("ROOT_DISABLE_GMI")) {
+      llvm::Optional<std::string> EnvDisGMI = llvm::sys::Process::GetEnv("ROOT_DISABLE_GMI");
+      if (EnvDisGMI.hasValue()) {
+         if (EnvEnGMI.hasValue())
+            ::Error("TCling__RegisterCxxModules",
```

Comment:
Why do we have 2 environment variable if they can never be set at the same time?

Reply:
Because if the platform is unsupported (by default that's windows) we should be able to enable the GMI there via an env variable. For the supported platforms we should be able to disable the GMI. In long run only one variable will be left -- ROOT_DISABLE_GMI

Reply:
I still don't "get" it ... If you are supposed to be carefully to not set one or the other (i.e. one does not supersede the other), then why can't you just set a single variable to true/false or on/off.

Reply:
I see your point. You want a single variable forcing the default -- like ROOT_ENABLE_GMI=1/0. Fair enough.